lib/masdb/data_map.ex

```diff
@@ -134,4 +141,23 @@ defmodule Masdb.Data.Map do
     |> Map.update!(key, &(%Masdb.Data.Val{since_ts: timestamp, value: &1}))
     |> normalize_to_vals(keys, timestamp)
   end
-end
+
+  defp get_rows(columns, data, filter, acc \\ []) 
+  defp get_rows(columns, [row | nexts], :all, acc) do
+    get_rows(columns, nexts, :all, acc ++ flatten_cols(columns, row.columns))
+  end
+
+  defp get_rows(_, [], :all, acc) do
+    acc
+  end
+
+  defp flatten_cols(columns, data, acc \\ [])
+  defp flatten_cols([row | nexts], data, acc) do
+    val = Map.get(data, row, %Masdb.Data.Val{since_ts: "", value: :nil})
+    flatten_cols(nexts, data, acc ++ [val.value])
```

Comment:
https://groups.google.com/d/msg/elixir-lang-talk/BR7wPpG9tzw/OP4QhIPHBwAJ

In short, we should go for a prepend `[val.value | acc]` and do a reverse on line 161.